### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by munichpavel** [munichpavel/fake-data-for-learning#14]
Using sphinx

**Comment by munichpavel** [comment]
Documentation a nice-to-have before packaging. Trying Sphinx + GitHub Pages.

Tutorial here: https://gisellezeno.com/tutorials/sphinx-for-python-documentation.html

**Comment by munichpavel** [comment]
This one mostly works out of the box for me:

**Comment by munichpavel** [comment]
To include project readme in docs landing page: https://stackoverflow.com/questions/46278683/include-my-markdown-readme-into-sphinx

**Comment by munichpavel** [comment]
The above is for link, which I got to work. To inject text, will try

https://muffinresearch.co.uk/selectively-including-parts-readme-rst-in-your-docs/

**Comment by munichpavel** [comment]
Nope, will pass on injection. Is blocked on some platforms anyway due to security risk.

**Comment by munichpavel** [comment]
Removing readme link, for posterity, the `readme_link.rst` that worked was

```
------
README
------

.. mdinclude:: ../../README.md

```

**Comment by munichpavel** [comment]
Plan to keep landing page as Jekyll GitHub page:  https://munichpavel.github.io/fake-data-for-learning that uses the GH readme, and then have documentation in Sphinx, likely in a differentiation repo due to non-jekyll-ness of Sphinx.

**Comment by munichpavel** [comment]
I have `sphinx-apidoc` working, but I don't like the standard headers it creates. This seems to be a reasonable fix:

https://stackoverflow.com/questions/21003122/sphinx-apidoc-section-titles-for-python-module-package-names

**Comment by munichpavel** [comment]
I use instead `sphinx-apidoc -fMe ../fake_data_for_learning -o source/api` as I want to keep headers.

**Comment by munichpavel** [comment]
API documentation good enough at commit https://github.com/munichpavel/fake-data-for-learning/commit/f6f00c5fc6f7e0aae28e5fc94dcc2eb5723e2438

Next step to work on hosting with GH Pages. I will first try a separate repo for docs hosting, so I can keep the landing page

https://munichpavel.github.io/fake-data-for-learning

as is.

Plan

* have build dir as environment variable in main project / repo. For me will be separate folder / repo but could be within same for other users
* Setup GitHub pages from docs repo via Sphinx
* Write bash scripts to automate updating of docs repo from main repo

**Comment by munichpavel** [comment]
Maybe plan first bullet not so good. I got it to work but with plan (ugly) html rending. Plus the "workaround" of having a `gh-pages` branch maybe isn't so bad, and can be incorporated in GitHub actions per here

https://stackoverflow.com/questions/57989790/using-github-actions-to-publish-documentation

**Comment by munichpavel** [comment]
Working now as planned: I was missing a required `.nojekyll`.

Next and last for now is to link up the various web homes. From sphinx docs seems most complicated:

https://gist.github.com/mgedmin/6052926

**Comment by munichpavel** [comment]
This is probably more useful, add link to GitHub source code within documentation

https://www.sphinx-doc.org/en/master/usage/extensions/linkcode.html

**Comment by munichpavel** [comment]
For possible later use or reference, 

https://github.com/sphinx-doc/sphinx/issues/701, how sphinx reads in CHANGES in root directory from docs/ folder.
